python/ray/autoscaler/v2/instance_manager/instance_manager.py

```diff
@@ -189,8 +194,10 @@ def get_valid_transitions() -> Dict[
                 # This happens when the cloud node first appears in the list of running
                 # cloud nodes from the cloud node provider.
                 Instance.ALLOCATED,
-                # Cloud provider fails to allocate one. Either as a timeout or
-                # the launch request fails immediately.
+                # Cloud provider failed to allocate a cloud node for the instance, retrying
```

Comment:
decided to make `ALLOCATION_FAILED` a terminal state rather than could be an intermediate state for easier querying of "which instances fail to acquire a cloud node".